parser.ts

```diff
@@ -322,6 +333,9 @@ export class Parser {
       const prevLineNo = this.lineNo;
       if (this.match(TokenKind.DecimalLiteral)) {
         this.lineNo = this.previous!.value as number;
+        this.line.lineNo = this.previous!.text;
+        this.line.trailingWhitespace = this.leadingWs;
```

Comment:
I think this and the following line are buggy, and that the special casing in advance() should handle what this is trying to do.